client/ayon_blender/plugins/publish/validate_frame_range.py

```diff
@@ -0,0 +1,84 @@
+import bpy
+import pyblish.api
+
+from ayon_core.pipeline import (
+    OptionalPyblishPluginMixin
+)
+from ayon_core.pipeline.publish import (
+    RepairAction,
+    ValidateContentsOrder,
+    PublishValidationError,
+    KnownPublishError
+)
+from ayon_blender.api.pipeline import get_frame_range
+
+
+class ValidateFrameRange(pyblish.api.InstancePlugin,
+                         OptionalPyblishPluginMixin):
+    """Validates the frame ranges.
+
+    This is an optional validator checking if the frame range on instance
+    matches the frame range specified for the folder.
+
+    It also validates render frame ranges of render layers.
+
+    Repair action will change everything to match the folder frame range.
+
+    This can be turned off by the artist to allow custom ranges.
+    """
+
+    label = "Validate Frame Range"
+    order = ValidateContentsOrder
+    families = ["render"]
+    hosts = ["blender"]
+    optional = True
+    actions = [RepairAction]
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            self.log.debug("Skipping Validate Frame Range...")
+            return
+
+        frame_range = get_frame_range(instance.data["taskEntity"])
+        scene = bpy.context.scene
+        inst_frame_start = scene.frame_start
+        inst_frame_end = scene.frame_end
+
+        if inst_frame_start is None or inst_frame_end is None:
+            raise KnownPublishError(
+                "Missing frame start and frame end on "
+                "instance to to validate."
+            )
+        frame_start_handle = frame_range["frameStartHandle"]
+        frame_end_handle = frame_range["frameEndHandle"]
+        errors = []
+        if frame_start_handle != inst_frame_start:
+            errors.append(
+                f"Start frame ({inst_frame_start}) on instance does not match " # noqa
+                f"with the start frame ({frame_start_handle}) set on the folder attributes. ")    # noqa
+        if frame_end_handle != inst_frame_end:
+            errors.append(
+                f"End frame ({inst_frame_end}) on instance does not match "
+                f"with the end frame ({frame_end_handle}) "
+                "from the folder attributes. ")
+
+        if errors:
+            bullet_point_errors = "\n".join(
+                "- {}".format(error) for error in errors
+            )
+            report = (
+                "Frame range settings are incorrect.\n\n"
+                f"{bullet_point_errors}\n\n"
+                "You can use repair action to fix it."
+            )
+            raise PublishValidationError(report, title="Frame Range incorrect")
+
+    @classmethod
+    def repair(cls, instance):
+        frame_range = get_frame_range()
+        frame_start_handle = frame_range["frameStartHandle"]
+        frame_end_handle = frame_range["frameEndHandle"]
+        scene = bpy.context.scene
+        scene.frame_start = frame_start_handle
+        scene.frame_end = frame_end_handle
```

Comment:
Is there a reason this behavior is different from [`set_frame_range`](https://github.com/ynput/ayon-blender/blob/3b893990d863afa5ecbf42658b3a8cc0d689b954/client/ayon_blender/api/pipeline.py#L257-L268)?

This includes the handles, the other doesn't. Why do they differ?

---

Also side note: Can we update `set_frame_range` (because currently it will fail to set a start or end frame of value 0. ;)

Looking at the logic it should be the same as:
```python
def set_frame_range(data: dict):
    """Set frame range
    
    Arguments:
        data (dict): Frame range and FPS
            to set the current scene to.
    """
    if not data:
        return

    scene = bpy.context.scene
    if "frameStart" in data:
        scene.frame_start = data["frameStart"]
    if "frameEnd" in data:
        scene.frame_end = data["frameEnd"]
    if "fps" in data:
        fps: float = data["fps"]
        scene.render.fps = round(fps)
        scene.render.fps_base = round(fps) / fps
```
Because the current code just sets what the scene is already set to if no data is provided (so we can just skip doing anything if not provided).